src/Sarif.Viewer.VisualStudio/Sarif/Run.extensions.cs

```diff
@@ -27,6 +29,20 @@ public static bool TryGetRule(this Run run, string ruleId, string ruleKey, out I
                     rule = run.Rules[ruleId];
                 }
             }
+            // No rule in log file, created "fake" rule
+            else
+            {
+                Byte[] ruleLookupBytes = Resource1.ruleLookup;
+                string ruleLookupText = Encoding.UTF8.GetString(ruleLookupBytes);
+                JObject metadata = JObject.Parse(ruleLookupText);
```

Comment:
Parsing the file for every defect/rule might be expensive. We should probably only read it from disk once (i.e. cache the data after reading the file.)